server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -429,7 +433,7 @@ public List<ColumnIdent> partitionedBy() {
     }
 
     public List<PartitionName> partitions() {
```

Comment:
I had more something like `getPartitions(ClusterState state)`  in mind (same for concreteIndices and concreteOpenIndices).

I also suspect that I `Supplier` isn't necessary - there shouldn't be any dependency other than the state and the `DocTableInfo` can contain the logic to resolve them from it, there's no need to host the logic in `DocTableInfoFactory` and pass it along. Everything in `IndexNameExpressionResolver` is static